homeassistant/components/media_player/vizio.py

```diff
@@ -0,0 +1,171 @@
+"""
+Vizio SmartCast TV support.
+
+Usually only 2016+ models come with SmartCast capabilities.
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.vizio/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.util as util
+from homeassistant.components.media_player import (
+    PLATFORM_SCHEMA,
+    SUPPORT_TURN_ON,
+    SUPPORT_TURN_OFF,
+    SUPPORT_SELECT_SOURCE,
+    SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_NEXT_TRACK,
+    SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_STEP,
+    MediaPlayerDevice
+)
+from homeassistant.const import (
+    STATE_UNKNOWN,
+    STATE_OFF,
+    STATE_ON,
+    CONF_NAME,
+    CONF_HOST,
+    CONF_ACCESS_TOKEN
+)
+from homeassistant.helpers import config_validation as cv
+
+REQUIREMENTS = ['git+https://github.com/vkorn/pyvizio.git'
+                '@master#pyvizio==0.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON = 'mdi:television'
+DEFAULT_NAME = 'Vizio SmartCast'
+DEVICE_NAME = 'Python Vizio'
+DEVICE_ID = 'pyvizio'
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
+SUPPORTED_COMMANDS = SUPPORT_TURN_ON | SUPPORT_TURN_OFF \
+                     | SUPPORT_SELECT_SOURCE \
+                     | SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK \
+                     | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_ACCESS_TOKEN): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the VizioTV media player platform."""
+    host = config.get(CONF_HOST)
+    if host is None:
+        _LOGGER.error('No host info')
+        return False
```

Comment:
That's covered by the configuration validation. `CONF_HOST` will never been `None`.